### PR TITLE
feat: copilot extension for prerelease

### PR DESCRIPTION
**Pull request by yuqizhou77**
This PR mainly focuses on the UI change of Copilot extension. Changes included:
- New UI for all copilot related features
- Walkthrough update 
- CLI questions for scaffolding and corresponding E2E change   

**TODO:**
- update feature flag name to "DEVELOP_COPILOT_EXTENSION". 
- work related to generator will be in a separate PR later.
- Corresponding change in VS when sending request 

image

image

if declarative Copilot:
image

if declarative Copilot with plugin or Plugin only:
image

E2E:
https://github.com/OfficeDev/teams-toolkit/actions/runs/10071203743

[User Story 28374074](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/28374074): Declarative Copilot in VSC

### CODE REVIEW

**Review by jayzhang — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by hund030 — DISMISSED** [review]

**Review by HuihuiWu-Microsoft — DISMISSED** [review]

**Review by huimiu — APPROVED** [review]

**Review by Alive-Fish — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -293,10 +293,12 @@
   "core.createProjectQuestion.projectType.officeAddin.framework.placeholder": "Select a framework",
   "core.createProjectQuestion.projectType.tab.detail": "Embed your own web content in Teams, Outlook, and the Microsoft 365 app",
   "core.createProjectQuestion.projectType.tab.title": "App Features Using a Tab",
+  "core.createProjectQuestion.projectType.copilotExtension.label": "Copilot Extension",
+  "core.createProjectQuestion.projectType.copilotExtension.title": "App Features using Copilot Extension",
+  "core.createProjectQuestion.projectType.copilotExtension.detail": "Create declarative copilot, API plugin, or both for customized guidance and integration.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:298]: Revised: Create declarative Copilot, API plugin, or both with MS Copilot orchestrator and LLM

Note: This was the latest content. Confirm with Junjie.

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -306,7 +308,7 @@
   "core.createProjectQuestion.projectType.copilotGroup.title": "Use Copilot",
   "core.createProjectQuestion.projectType.createGroup.title": "Create",
   "core.createProjectQuestion.projectType.declarativeCopilot.label": "Declarative Copilot",
-  "core.createProjectQuestion.projectType.declarativeCopilot.title": "Author a Declarative Copilot",
+  "core.createProjectQuestion.projectType.declarativeCopilot.detail": "Craft your own Copilot by declaring instructions, actions, and knowledge to customize Copilot for Microsoft 365.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:311]: Revised: Create your own copilot by declaring instructions, actions, & knowledge to suit your needs.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -293,10 +293,12 @@
...
   "core.createProjectQuestion.projectType.tab.title": "App Features Using a Tab",
+  "core.createProjectQuestion.projectType.copilotExtension.label": "Copilot Extension",
+  "core.createProjectQuestion.projectType.copilotExtension.title": "App Features using Copilot Extension",
+  "core.createProjectQuestion.projectType.copilotExtension.detail": "Create declarative copilot, API plugin, or both for customized guidance and integration.",
   "core.createProjectQuestion.projectType.copilotPlugin.detail": "Create a plugin to extend Microsoft Copilot for Microsoft 365 using your APIs",
   "core.createProjectQuestion.projectType.copilotPlugin.label": "API Plugin",
-  "core.createProjectQuestion.projectType.copilotPlugin.title": "API Plugin",
-  "core.createProjectQuestion.projectType.copilotPlugin.placeholder": "Select an option",
+  "core.createProjectQuestion.projectType.copilotExtension.placeholder": "Select an option",
   "core.createProjectQuestion.projectType.customCopilot.detail": "Build intelligent chatbot in Microsoft Teams easily using Teams AI Library",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:302]: Revised: Build intelligent chatbot with Teams AI Library where you manage orchestration and provide your own LLM.

Note: Confirm with Junjie.

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:298]: @MuyangAmigo are you OK the revised string proposed by Supriya?

File: packages/fx-core/resource/package.nls.json (hunk 3 above)
Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:302]: @MuyangAmigo are you OK the revised string proposed by Supriya?

Comment (MuyangAmigo) [reply packages/fx-core/resource/package.nls.json:302]: Yes this was discussed in an other thread. Thanks folks!

File: packages/fx-core/resource/package.nls.json (hunk 1 above)
Comment (MuyangAmigo) [reply packages/fx-core/resource/package.nls.json:298]: Yes, a minor thing is "MS" -> "Microsoft". Thanks!

**Review by supkasar — APPROVED** [review]

**Review by hund030 — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Review by huimiu — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**